prov/tcp/src/xnet_domain.c

```diff
@@ -226,12 +489,18 @@ int xnet_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 	if (ret)
 		return ret;
 
+	if (info->ep_attr->type == FI_EP_RDM &&
+	    info->domain_attr->threading == FI_THREAD_COMPLETION)
+		return xnet_domain_multiplex_open(fabric_fid, info, domain_fid,
+						  context);
+
 	domain = calloc(1, sizeof(*domain));
 	if (!domain)
 		return -FI_ENOMEM;
 
 	ret = ofi_domain_init(fabric_fid, info, &domain->util_domain, context,
-			      OFI_LOCK_NONE);
+			      info->domain_attr->threading == FI_THREAD_SAFE ?
```

Comment:
I can't remember why we added this - was it a separate optimization or was it needed for these changes?

Reply:
I think this should actually be removed.   think it's left over from some earlier changes and likely has no real affect (since our new changes only apply to FI_THREAD_COMPLETION and this code is in the non-FI_THREAD_COMPLETION path, it should remain unchanged)